dbg.py

```diff
@@ -18,94 +18,79 @@ def __init__(self):
-		res.ParseFromString(self._sock.recv(4096))
-		if res.type != Response.OK:
-			raise XboxError(res.msg)
-		return res
+	def ensure_recv(self, size):
+		#FIXME: Block
+		data = bytes()
+		while size > 0:
+			chunk = self._sock.recv(size)
+			data = data + chunk
+			size -= len(chunk)
+		return data
 
 	def info(self):
 		"""Get system info"""
-		req = Request()
-		req.type = Request.SYSINFO
-		return self._send_simple_request(req).info
+		request_buffer = struct.pack("<B", 0)
+		print(len(request_buffer))
+		self._sock.send(request_buffer)
+		response_buffer = self.ensure_recv(4)
+		response_data = struct.unpack("<I", response_buffer)
+		result = {}
+		result['tick_count'] = response_data[0]
+		return result
+
+	def debug_print(self, message):
+		"""Print a debug string to the screen"""
+		message_buffer = message.encode('ascii')
+		request_buffer = struct.pack("<BH", 1, len(message_buffer)) + message_buffer
+		self._sock.send(request_buffer)
 
 	def reboot(self):
 		"""Reboot the system"""
-		msg = Request()
-		msg.type = Request.REBOOT
-		self._sock.send(msg.SerializeToString())
+		request_buffer = struct.pack("<B", 2)
+		self._sock.send(request_buffer)
 
 	def malloc(self, size):
 		"""Allocate memory on the target"""
-		req = Request()
-		req.type = Request.MALLOC
-		req.size = size
-		return self._send_simple_request(req).address
+		request_buffer = struct.pack("<BI", 3, size)
+		self._sock.send(request_buffer)
+		response_buffer = self.ensure_recv(4)
+		response_data = struct.unpack("<I", response_buffer)
+		return response_data[0]
 
-	def free(self, addr):
+	def free(self, address):
 		"""Free memory on the target"""
-		req = Request()
-		req.type = Request.FREE
-		req.address = addr
-		return self._send_simple_request(req)
+		request_buffer = struct.pack("<BI", 4, address)
+		self._sock.send(request_buffer)
 
-	def mem_read(self, addr, size):
+	def mem_read(self, address, size):
 		"""read memory"""
-		req = Request()
-		req.type = Request.MEM_READ
-		req.size = size
-		req.address = addr
-		return self._send_simple_request(req).data
+		request_buffer = struct.pack("<BII", 5, address, size)
+		self._sock.send(request_buffer)
+		response_buffer = self.ensure_recv(size)
+		return response_buffer
 
-	def mem_write(self, addr, data):
+	def mem_write(self, address, data):
 		"""write memory"""
-		req = Request()
-		req.type = Request.MEM_WRITE
-		req.data = data
-		req.address = addr
-		return self._send_simple_request(req)
-
-	def debug_print(self, string):
-		"""Print a debug string to the screen"""
-		req = Request()
-		req.type = Request.DEBUG_PRINT
-		req.msg = string
-		return self._send_simple_request(req)
-
-	def show_debug_screen(self):
-		"""Show the debug screen"""
-		req = Request()
-		req.type = Request.SHOW_DEBUG_SCREEN
-		return self._send_simple_request(req)
-
-	def show_front_screen(self):
-		"""Show the front screen"""
-		req = Request()
-		req.type = Request.SHOW_FRONT_SCREEN
-		return self._send_simple_request(req)
```

Comment:
I've removed these `pb_show_...` functions, as the buffer isn't even allocated. If the user only wants to change the framebuffer, they can do it differently.